DESIGN_PROPOSAL.md

```diff
@@ -0,0 +1,123 @@
+   - *agogiczna* - liczba zmian w tempie (metazdarzenie `FF 51 03`);
+     na poziomie MIDI "ogólne" zmiany tempa są nieodróżnialne od chwilowych przyspieszeń lub spowolnień (accel., rit., fermata), ale można próbować je sklasyfikować ze względu na długość oraz umiejscowienie,
+   - *rytmiczna* - liczba zmian metrum (metazdarzenie `FF 58 04`),
+     pytaniem pozostaje, czy istnieją modele generatywne, które zmiany metrum wspierają.
+
+Opisane wyżej metryki można wyliczyć dla istniejących utworów reprezentujacych różne gatunki (niżej opisany zbiór danych nr 1) i na podstawie wyników próbować klasyfikować utwory wygenerowane (zbiór danych nr 2).
+Podobnie jak w algorytmie "Inception Score", wysoka niejednoznaczność klasyfikacji pojedynczego utworu mogłaby oznaczać słabą "ostrość" reprezentowanego przez niego gatunku, a więc słabą jakość.
+Uważać trzeba na przypadki szczególne - mieszanie gatunków w obrębie pojedynczego utworu, gatunki pośrednie lub niedookreślone.
+
+Ponadto, możemy próbować przekształcić powyższe metryki na dwustopniowe lub trzystopniowe miary subiektywne (jakość dobra - zła oraz brak wpływu metryki na ocenę).
+Jesteśmy w stanie dokonać badań statystycznych polegających na ocenie wycinka zbiorów danych w obrębie naszego zespołu.
+Badania takie miałyby na celu określenie akceptowalnych i nieakceptowalnych zakresów wartości wyznaczonych metryk.
+
+## Zbiory danych
+
+Zgrubny podział zbiorów danych:
+
+1. istniejące utwory w MIDI (podzbiór katalogu oferowanego w ramach MusPy),
+2. utwory w MIDI wygenerowane z użyciem istniejących, wytrenowanych modeli uczenia maszynowego (technologia GAN [^dong2017] lub Transformer [^huang2018]).
+
+Zbiór nr 1 możemy poszerzyć o własnoręcznie spreparowane dane reprezentujące utwory słabej jakości.
+
+Dla obu zbiorów wyznaczymy wartości metryk opracowanych przez nas, jak i obecnych już w MusPy (celem porównania).
+
+## Planowany zakres eksperymentów
+
+1. Zbadamy graniczne oraz umiarkowane wartości niektórych metryk poprzez spreparowanie plików MIDI o odpowiedniej charakterystyce.
+   W przypadku łańcuchów Markova zależy nam na wartościach dla braku powtarzalności, stuprocentowej powtarzalności oraz umiarkowanej powtarzalności w postaci znanych form okresowych utworów (ABA, ABA', AABA, ABAC, itd.)
+2. Dla wybranych zbiorów danych porównamy metryki z MusPy z naszymi.
+   Sprawdzimy, czy są od siebie zależne, tzn. czy kierunek zmiany obu metryk jest skorelowany.
+3. (opcjonalnie) Zbadamy, jak mają się wartości metryk na wybranych utworach do naszych odczuć.
+   Zrobimy to w sposób porównawczy, prezentując badanej osobie dwa utwory o różnych wartościach metryk.
+   Potencjalnie wykorzystamy do oceny skalę Likerta z ogólnymi stwierdzeniami postaci "Utwór nr 1 brzmi lepiej niż utwór nr 2".
+
+## Planowany stack technologiczny
+
+Język programowania: Python  
+Język opisu: Markdown
+
+Docelowy format: biblioteka z przykładem użycia w postaci narzędzia konsolowego.
```

Comment:
```suggestion
Docelowy format: biblioteka oraz proste narzędzie konsolowe.
```

Reply:
nie o to mi chodziło?

przykłady w formie kilku plików `py` które uruchamiają przykładowy kod

narzędzie konsolowe tylko komplikuje wykorzystanie bo nagle trzeba odpalać konsole i podawać jakieś wymyślone argumenty

plik .py sobie odpalasz w VSC i klikasz przycisk i działa
kod można łatwo opisać i czytać bo to jest najprostszy kawałek pythona możliwy

Reply:
```suggestion
Docelowy format: biblioteka.
```